sdk/cosmos/azure_data_cosmos/src/pipeline/mod.rs

```diff
@@ -29,24 +30,29 @@ use crate::{
 pub struct CosmosPipeline {
     pub endpoint: Url,
     pipeline: azure_core::http::Pipeline,
+    retry_handler: BackOffRetryHandler,
 }
 
 impl CosmosPipeline {
     pub fn new(
         endpoint: Url,
         auth_policy: AuthorizationPolicy,
-        client_options: ClientOptions,
+        mut client_options: ClientOptions,
     ) -> Self {
+        client_options.retry = RetryOptions::none();
```

Comment:
You should probably capture what the user specified first in case any of that is useful in your retry policy. Don't completely ignore any user options they set; barring that, you might even err if they set anything so they know why their options are getting ignored.